kclvm/tools/src/query/override.rs

```diff
@@ -138,44 +326,182 @@ impl<'ctx> MutSelfMutWalker<'ctx> for OverrideTransformer {
+            },
+            None => "".to_string(),
+        }
     }
-    pub(crate) fn _find_schema_config_and_repalce(
+}
+
+/// OverrideNode is used to place different AST configuration syntax structures.
+#[derive(Debug)]
+enum OverrideNode<'a> {
+    Schema(&'a mut ast::SchemaExpr),
+    Config(&'a mut ast::ConfigExpr),
+}
+
+impl OverrideTransformer {
+    /// Lookup schema config all fields and replace if it is matched with the override spec,
+    /// return whether is found a replaced one.
+    fn lookup_schema_config_and_replace(&mut self, schema_expr: &mut ast::SchemaExpr) -> bool {
+        let paths = self.get_schema_config_field_paths(schema_expr);
+        match paths.iter().position(|r| r == &self.field_path) {
+            Some(pos) => {
+                let mut config = OverrideNode::Schema(schema_expr);
+                self.replace_node_with_path(&mut config, &paths[pos]);
+                true
+            }
+            None => false,
+        }
+    }
+
+    /// Replace AST node with path including schema_expr and config_expr.
+    fn replace_node_with_path(&mut self, config: &mut OverrideNode, path: &str) {
+        // Do not replace empty path parts on the config expression.
+        if path.is_empty() {
+            return;
+        }
+        // Split a path into multiple parts. `a.b.c` -> ["a", "b", "c"]
+        let parts = path.split('.').collect::<Vec<&str>>();
+        for i in 0..parts.len() {
+            match config {
+                OverrideNode::Schema(schema_expr) => {
+                    if let ast::Expr::Config(config_expr) = &mut schema_expr.config.node {
+                        self.replace_config_expr_with_path_parts(config_expr, &parts, i);
+                    }
+                }
+                OverrideNode::Config(config_expr) => {
+                    self.replace_config_expr_with_path_parts(config_expr, &parts, i);
+                }
+            }
+        }
+    }
+
+    /// Replace AST config expr with one part of path.
+    fn replace_config_expr_with_path_parts(
         &mut self,
-        _schema_config: &mut ast::SchemaExpr,
-        _field_path: &str,
-        _value: &ast::NodeRef<ast::Expr>,
-    ) -> bool {
-        false
+        config_expr: &mut ast::ConfigExpr,
+        parts: &[&str],
+        start_index: usize,
+    ) {
+        // Do not replace empty path parts and out of index parts on the config expression.
+        if parts.is_empty() || start_index >= parts.len() {
+            return;
+        }
+        let part = parts[start_index];
+        let mut delete_index_set = HashSet::new();
+        for (i, item) in config_expr.items.iter_mut().enumerate() {
+            let path = self.get_path_from_key(&item.node.key);
+            if path == part {
+                match self.action {
+                    ast::OverrideAction::CreateOrUpdate => {
+                        // When the last part of the path is successfully recursively matched, 
```

Comment:
Reorg the comments, desc 'path == part' and 'start_index == 0 && parts.len() == 1' clearly

Reply:
Haved added clearer comments